src/UnitfulRecipes.jl

```diff
@@ -42,32 +80,45 @@ function resolve_axis!(attr, arr::AbstractArray{T}, axis::Int) where {T<:Quantit
 
     # get axis label and append unit
     key = key_label(axis)
-    if haskey(attr, key)
-        attr[key] = string(attr[key], " ($(string(u)))")
-    else
-        attr[key] = string(u)
+    ckey = (axis==3) ? :colorbar_title : key
+    ustr = string(u)
+    if haskey(attr, key) # Append unit (only once) if axis label exists
+        i = findlast(ustr, attr[key])
```

Comment:
Why not use `occursin(ustr, attr[key])`?

Reply:
E.g. in case you have data in meters and you label it with a string that contains an "m".